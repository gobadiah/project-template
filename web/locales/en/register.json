{
  "title": "Register",
  "Create your account": "Create your account",
  "Signin": "Signin"
}
