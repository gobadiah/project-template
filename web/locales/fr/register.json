{
  "title": "Créer un compte",
  "Create your account": "Créer un compte",
  "Signin": "Se connecter"
}
