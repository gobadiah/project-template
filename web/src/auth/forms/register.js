import { reduxForm } from 'redux-form';
import React from 'react';

import {
  BirthdayField,
  EmailField,
  FirstNameField,
  LastNameField,
  PasswordField,
} from '~/components/forms/fields';
import { Form } from '~/components/forms';

import { PureComponent } from '~/components/base';

class RegisterForm extends PureComponent {
  render() {
    return (
      <Form {...this.props}>
        <EmailField key='email' />
        <PasswordField
          key='password'
          autoComplete='new-password'
        />
        <PasswordField
          key='confirm_password'
          name='confirm_password'
          label='Password confirmation'
          autoComplete='new-password'
        />
        <FirstNameField key='first_name' />
        <LastNameField key='last_name' />
        <BirthdayField key='birthday' />
      </Form>
    );
  }
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
