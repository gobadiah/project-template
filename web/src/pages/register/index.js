import React from 'react';

import { Page } from '~/components/base';
import hoc from '~/hoc';

import { RegisterForm } from '~/auth';

class Register extends Page {
  render() {
    const { register } = this.props;
    return (
      <RegisterForm onSubmit={register} />
    );
  }
}

export default hoc('register')(Register);
