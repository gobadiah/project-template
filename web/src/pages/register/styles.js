import { css } from 'react-emotion';

export const containerClassName = css`
  width: 640px;
  height: 1106px;
`;
