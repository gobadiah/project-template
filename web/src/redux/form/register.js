import React from 'react';

import { reduxForm } from 'redux-form';

import {
  BirthdayField,
  EmailField,
  FirstNameField,
  LastNameField,
  PasswordField,
} from './fields';
import { BaseForm } from './utils';

class RegisterForm extends BaseForm {
  getFields = () => [
    <EmailField key='email' />,
    <PasswordField
      key='password'
      autoComplete='new-password'
    />,
    <PasswordField
      key='confirm_password'
      name='confirm_password'
      label='Password confirmation'
      autoComplete='new-password'
    />,
    <FirstNameField key='first_name' />,
    <LastNameField key='last_name' />,
    <BirthdayField key='birthday' />,
  ];
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
