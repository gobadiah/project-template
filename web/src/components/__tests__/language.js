import 'jsdom-global/register';

import React from 'react';
import { shallow } from 'enzyme';

import Language, { Lang } from '../language';

const setup = (component, props = {}) => {
  const i18n = {
    changeLanguage: jest.fn(),
    t: jest.fn(key => key),
  };
  const wrapper = shallow(React.createElement(component, props), { context: { i18n, t: i18n.t } });
  return {
    wrapper,
    i18n,
  };
};

describe('Language', () => {
  it('should match snapshot', () => {
    const { wrapper } = setup(Language);
    expect(wrapper).toMatchSnapshot();
  });

  it('should contains two buttons', () => {
    const { wrapper } = setup(Language);
    expect(wrapper.find(Lang)).toHaveLength(2);
  });
});

describe('Lang', () => {
  const setupLang = () => setup(Lang, { lng: 'fr', name: 'french', img: 'img' });
  it('should match snapshot', () => {
    const { wrapper } = setupLang();
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain an input', () => {
    const { wrapper, i18n } = setupLang();
    expect(wrapper.find('input')).toHaveLength(1);
    wrapper.find('input').at(0).prop('onClick')();
    expect(i18n.changeLanguage.mock.calls).toHaveLength(1);
    expect(i18n.changeLanguage.mock.calls[0]).toEqual(['fr']);
  });
});
