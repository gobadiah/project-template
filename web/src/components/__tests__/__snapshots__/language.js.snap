// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lang should match snapshot 1`] = `
<div>
  <input
    onClick={[Function]}
    type="button"
    value="Switch to french"
  />
  <img
    alt=""
    src="img"
  />
</div>
`;

exports[`Language should match snapshot 1`] = `
<div>
  <Lang
    i18n={
      Object {
        "changeLanguage": [MockFunction],
        "t": [MockFunction],
      }
    }
    img="/static/build/329f5c89d4f13938971b9b0d8d35f296.png"
    lng="fr"
    name="french"
    t={[MockFunction]}
  />
  <Lang
    i18n={
      Object {
        "changeLanguage": [MockFunction],
        "t": [MockFunction],
      }
    }
    img="/static/build/1b46775dab5a7245b5969a5aa84e0dff.png"
    lng="en"
    name="english"
    t={[MockFunction]}
  />
</div>
`;
